From 38425f6beaa496c92470ed5e51740523c35def36 Mon Sep 17 00:00:00 2001
From: Alexandre TORGUE <alexandre.torgue@st.com>
Date: Thu, 16 May 2024 11:12:20 +0200
Subject: [PATCH 596/954] feat(stm32mp23-fdts): add RCC and RIF configuration
 for stm32mp235f-dk

Add minimal RCC clocktree and RISAF configuration.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 9010d81d53befab1d6e9ce745d11ad9a61b98714 feat(stm32mp23-fdts): add RCC and RIF configuration for stm32mp235f-dk]
Change-Id: I0c2f280ef832c3201fdc7c84522854c23dae48e0
Signed-off-by: Alexandre TORGUE <alexandre.torgue@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/380125
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 fdts/stm32mp235f-dk-ca35tdcid-fw-config.dtsi | 34 +++++++
 fdts/stm32mp235f-dk-ca35tdcid-rcc.dtsi       | 98 ++++++++++++++++++++
 fdts/stm32mp235f-dk-fw-config.dts            |  1 +
 3 files changed, 133 insertions(+)
 create mode 100644 fdts/stm32mp235f-dk-ca35tdcid-fw-config.dtsi
 create mode 100644 fdts/stm32mp235f-dk-ca35tdcid-rcc.dtsi

diff --git a/fdts/stm32mp235f-dk-ca35tdcid-fw-config.dtsi b/fdts/stm32mp235f-dk-ca35tdcid-fw-config.dtsi
new file mode 100644
index 0000000000..f81f2d6d8b
--- /dev/null
+++ b/fdts/stm32mp235f-dk-ca35tdcid-fw-config.dtsi
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ */
+
+/*
+ * stm32mp235f tf-a firmware config
+ * Project : open
+ * Generated by XLmx tool version 2.2 - 5/16/2024 11:06:21 AM
+ */
+
+/ {
+	dtb-registry {
+		soc_fw-config {
+			load-address = <0x0 0x81fc0000>;
+			max-size = <0x40000>;
+		};
+		tos_fw {
+			load-address = <0x0 0x82000000>;
+			max-size = <0x2000000>;
+		};
+	};
+
+	st-mem-firewall {
+		bl31_context: bl31-context@81fc0000 {
+			reg = <0x0 0x81fc0000 0x0 0x40000>;
+			st,protreg = <RISAFPROT(RISAF_REG_ID(7), RIF_CID0_BF|RIF_CID1_BF, RIF_CID0_BF|RIF_CID1_BF, RIF_CID1_BF, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
+		};
+		op_tee: op-tee@82000000 {
+			reg = <0x0 0x82000000 0x0 0x2000000>;
+			st,protreg = <RISAFPROT(RISAF_REG_ID(8), RIF_CID0_BF|RIF_CID1_BF, RIF_CID0_BF|RIF_CID1_BF, 0, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
+		};
+	};
+};
diff --git a/fdts/stm32mp235f-dk-ca35tdcid-rcc.dtsi b/fdts/stm32mp235f-dk-ca35tdcid-rcc.dtsi
new file mode 100644
index 0000000000..79f405c50a
--- /dev/null
+++ b/fdts/stm32mp235f-dk-ca35tdcid-rcc.dtsi
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Loic Pallardy loic.pallardy@foss.st.com for STMicroelectronics.
+ */
+
+/*
+ * stm32mp235f Clock tree device tree configuration
+ * Project : open
+ * Generated by XLmx tool version 2.2 - 5/16/2024 11:06:20 AM
+ */
+
+&clk_hse {
+	clock-frequency = <40000000>;
+};
+
+&clk_hsi {
+	clock-frequency = <64000000>;
+};
+
+&clk_lse {
+	clock-frequency = <32768>;
+};
+
+&clk_lsi {
+	clock-frequency = <32000>;
+};
+
+&clk_msi {
+	clock-frequency = <16000000>;
+};
+
+&rcc {
+	st,busclk = <
+		DIV_CFG(DIV_LSMCU, 1)
+		DIV_CFG(DIV_APB1, 0)
+		DIV_CFG(DIV_APB2, 0)
+		DIV_CFG(DIV_APB3, 0)
+		DIV_CFG(DIV_APB4, 0)
+		DIV_CFG(DIV_APBDBG, 0)
+	>;
+
+	st,flexgen = <
+		FLEXGEN_CFG(0, XBAR_SRC_PLL4, 0, 2)
+		FLEXGEN_CFG(1, XBAR_SRC_PLL4, 0, 5)
+		FLEXGEN_CFG(2, XBAR_SRC_PLL4, 0, 1)
+		FLEXGEN_CFG(4, XBAR_SRC_PLL4, 0, 3)
+		FLEXGEN_CFG(5, XBAR_SRC_PLL4, 0, 2)
+		FLEXGEN_CFG(8, XBAR_SRC_HSI_KER, 0, 0)
+		FLEXGEN_CFG(48, XBAR_SRC_PLL5, 0, 3)
+		FLEXGEN_CFG(51, XBAR_SRC_PLL4, 0, 5)
+		FLEXGEN_CFG(52, XBAR_SRC_PLL4, 0, 5)
+		FLEXGEN_CFG(58, XBAR_SRC_HSE_KER, 0, 1)
+		FLEXGEN_CFG(63, XBAR_SRC_PLL4, 0, 2)
+	>;
+
+	st,kerclk = <
+		MUX_CFG(MUX_USB2PHY1, MUX_USB2PHY1_FLEX57)
+		MUX_CFG(MUX_USB2PHY2, MUX_USB2PHY2_FLEX58)
+	>;
+
+	pll1: st,pll-1 {
+		st,pll = <&pll1_cfg_1200Mhz>;
+
+		pll1_cfg_1200Mhz: pll1-cfg-1200Mhz {
+			cfg = <30 1 1 1>;
+			src = <MUX_CFG(MUX_MUXSEL5, MUXSEL_HSE)>;
+		};
+	};
+
+	pll2: st,pll-2 {
+		st,pll = <&pll2_cfg_600Mhz>;
+
+		pll2_cfg_600Mhz: pll2-cfg-600Mhz {
+			cfg = <30 1 1 2>;
+			src = <MUX_CFG(MUX_MUXSEL6, MUXSEL_HSE)>;
+		};
+	};
+
+	pll4: st,pll-4 {
+		st,pll = <&pll4_cfg_1200Mhz>;
+
+		pll4_cfg_1200Mhz: pll4-cfg-1200Mhz {
+			cfg = <30 1 1 1>;
+			src = <MUX_CFG(MUX_MUXSEL0, MUXSEL_HSE)>;
+		};
+	};
+
+	pll5: st,pll-5 {
+		st,pll = <&pll5_cfg_532Mhz>;
+
+		pll5_cfg_532Mhz: pll5-cfg-532Mhz {
+			cfg = <133 5 1 2>;
+			src = <MUX_CFG(MUX_MUXSEL1, MUXSEL_HSE)>;
+		};
+	};
+};
+
diff --git a/fdts/stm32mp235f-dk-fw-config.dts b/fdts/stm32mp235f-dk-fw-config.dts
index f2b68e8798..0862bbd04b 100644
--- a/fdts/stm32mp235f-dk-fw-config.dts
+++ b/fdts/stm32mp235f-dk-fw-config.dts
@@ -4,3 +4,4 @@
  */
 
 #include "stm32mp23-fw-config.dtsi"
+#include "stm32mp235f-dk-ca35tdcid-fw-config.dtsi"
-- 
2.53.0

